### PR TITLE
Form field bootstrap templates

### PR DESCRIPTION
**Pull request by darioalbanesi**
**This pull request fixes the following issues**:

As bootstrap uses several different classes for each form field, we need to create a templates with the corresponding classes for every (in aldryn-forms) available form field.

e.g. Checkbox, Radiobutton, File Upload, and some more.

Please also check twitter bootstrap doc:
http://getbootstrap.com/css/#forms-controls

- [x] WAI Checked ([chrome extension](https://chrome.google.com/webstore/detail/accessibility-developer-t/fpkknkljclfencbdbgkenhalefipecmb?hl=en))
- [x] Validity Checked ([chrome extension]( https://chrome.google.com/webstore/detail/validity/bbicmjjbohdfglopkidebfccilipgeif?hl=en))

Additional Information:

- All available fields have been checked and tested on a new aldryn project (locally). All tests passed.

- ~~Checkboxes should have the possibility to define whether they are checked or not.~~

- ~~Radiobuttons should have the possibility to select the default.~~

- ~~As [select2](https://select2.github.io/) is used on some projects, we might give some options for the admin to set, e.g. minimum-results-for-search="-1" to hide the the search-box. Just a nice to have.~~

### CODE REVIEW

**Comment by czpython** [comment]
@darioalbanesi please rename the template to ```booleanfield.html```, the reason being that the field type is boolean instead of checkbox. you can see it [here](https://github.com/aldryn/aldryn-forms/blob/master/aldryn_forms/cms_plugins.py#L541)

File: aldryn_forms/boilerplates/bootstrap3/templates/aldryn_forms/fields/booleanfield.html [hunk 1]
```diff
@@ -0,0 +1,17 @@
+{% load aldryn_forms_tags i18n %}
+
+<div class="checkbox{% if field.field.required %} form-required{% endif %}{% if field.errors %} has-error{% endif %}">
+    {% block form_field_label %}
+    <label>
+        {% block form_field %}
+        <input type="checkbox" value="">
```
Comment (FinalAngel) [inline aldryn_forms/boilerplates/bootstrap3/templates/aldryn_forms/fields/booleanfield.html]: you need to pre-fetch the value there and the "name" is missing

File: aldryn_forms/boilerplates/bootstrap3/templates/aldryn_forms/fields/emailfield.html [hunk 2]
```diff
@@ -0,0 +1,15 @@
+{% load aldryn_forms_tags i18n %}
+
+<div class="form-group{% if field.field.required %} form-required{% endif %}{% if field.errors %} has-error{% endif %}">
+    {% block form_field_label %}
+        <label>{{ field.label }}{% if field.field.required %}<abbr title="{% trans "Required field" %}">*</abbr>{% endif %}</label>
+    {% endblock form_field_label %}
+    {% block form_field %}
+        <input type="email" class="form-control" />
```
Comment (FinalAngel) [inline aldryn_forms/boilerplates/bootstrap3/templates/aldryn_forms/fields/emailfield.html]: also name missing and `/>` is not required anymore :)

File: aldryn_forms/boilerplates/bootstrap3/templates/aldryn_forms/fields/textareafield.html [hunk 3]
```diff
@@ -0,0 +1,12 @@
+{% load aldryn_forms_tags i18n %}
+
+<div class="form-group{% if field.field.required %} form-required{% endif %}{% if field.errors %} has-error{% endif %}">
+    {% block form_field %}
+    <textarea class="form-control" rows="3"></textarea>
```
Comment (FinalAngel) [inline aldryn_forms/boilerplates/bootstrap3/templates/aldryn_forms/fields/textareafield.html]: rows and cols should probably be provided by the plugin itself

File: aldryn_forms/boilerplates/bootstrap3/templates/aldryn_forms/fields/filefield.html [hunk 4]
```diff
@@ -0,0 +1,20 @@
+{% load aldryn_forms_tags i18n %}
+
+{% if field %}
+    <div class="form-group{% if field.field.required %} form-required{% endif %}{% if field.errors %} has-error{% endif %}">
+        {% block form_field_label %}
+            <label>{{ field.label }}{% if field.field.required %}<abbr title="{% trans "Required field" %}">*</abbr>{% endif %}</label>
```
Comment (FinalAngel) [inline aldryn_forms/boilerplates/bootstrap3/templates/aldryn_forms/fields/filefield.html]: the label needs to be always associated with the input through ``<label for="field-input-1234">` and `<input id="field-input-1234">` for all cases

**Comment by FinalAngel** [comment]
- Check WAI
- Check Validation

File: aldryn_forms/boilerplates/bootstrap3/templates/aldryn_forms/fields/filefield.html [hunk 5]
```diff
@@ -0,0 +1,20 @@
+{% load aldryn_forms_tags i18n %}
+
+{% if field %}
+    <div class="form-group{% if field.field.required %} form-required{% endif %}{% if field.errors %} has-error{% endif %}">
+        {% block form_field_label %}
+            <label for="{{ field.auto_id }}">{{ field.label }}{% if field.field.required %}<abbr title="{% trans "Required field" %}">*</abbr>{% endif %}</label>
+        {% endblock form_field_label %}
+        {% block form_field %}
+            <input type="file" class="form-control" name="{{ field.name }}">
```
Comment (FinalAngel) [inline aldryn_forms/boilerplates/bootstrap3/templates/aldryn_forms/fields/filefield.html]: how is the input associated with the label?

**Comment by FinalAngel** [comment]
Whats the status here @darioalbanesi ?

**Comment by darioalbanesi** [comment]
@FinalAngel waiting for review. I adapted your comments. Will post it again on hipchat.

**Comment by ruslanmalogulko** [comment]
LGTM for me

File: aldryn_forms/boilerplates/bootstrap3/templates/aldryn_forms/fields/selectfield.html [hunk 6]
```diff
@@ -0,0 +1,30 @@
...
+
+{% if field %}
+    <div class="form-group{% if field.field.required %} form-required{% endif %}{% if field.errors %} has-error{% endif %}">
+        {% block form_field_label %}
+            <label for="{{ field.auto_id }}">{{ field.label }}{% if field.field.required %}<abbr title="{% trans "Required field" %}">*</abbr>{% endif %}</label>
+        {% endblock form_field_label %}
+        {% block form_field %}
+            <select  class="form-control" name="{{ field.name }}" id="{{ field.auto_id }}">
+            {% comment %}
+                NOTE: Need to give the possibility to enable/disable te search field in the select dropdown.
```
Comment (sergii-kulakovskyi) [inline aldryn_forms/boilerplates/bootstrap3/templates/aldryn_forms/fields/selectfield.html]: typo "te" -> "the". The same in multipleselectfield.html.
+ better to indent right this comment.
P.S.: yes, it doesn't refers to the code in general.

File: aldryn_forms/boilerplates/bootstrap3/templates/aldryn_forms/fields/multipleselectfield.html [hunk 7]
```diff
@@ -0,0 +1,31 @@
...
+                {% for choice in field.field.choices %}
+                    <option>
+                        {{ choice.1 }}
+                    </option>
+                {% endfor %}
+            </select>
+        {% endblock form_field %}
+        {% block form_field_errors %}
+            {% if field.errors %}
+                {% for error in field.errors %}<span class="text-danger">{{ error }}</span>{% endfor %}
```
Comment (sergii-kulakovskyi) [inline aldryn_forms/boilerplates/bootstrap3/templates/aldryn_forms/fields/multipleselectfield.html]: Guidelines says you have to start code after `{% if %}`, `{% forloop %}`, `{% block %}` with new line and indent.
Please check if it's right for all templates, and correct if it's not harm readability.

**Comment by vxsx** [comment]
aside from minor typo LGTM

File: aldryn_forms/boilerplates/bootstrap3/templates/aldryn_forms/fields/booleanfield.html [hunk 8]
```diff
@@ -0,0 +1,21 @@
+{% load aldryn_forms_tags i18n %}
+
+{% if field %}
+    <div class="checkbox{% if field.field.required %} form-required{% endif %}{% if field.errors %} has-error{% endif %}">
+        {% block form_field_label %}
+        <label for="{{ field.auto_id }}">
```
Comment (FinalAngel) [inline aldryn_forms/boilerplates/bootstrap3/templates/aldryn_forms/fields/booleanfield.html]: I think here it also needs an indent?

File: aldryn_forms/boilerplates/bootstrap3/templates/aldryn_forms/fields/radioselectfield.html [hunk 9]
```diff
@@ -0,0 +1,26 @@
+{% load aldryn_forms_tags i18n %}
+
+{% if field %}
+    {% for choice in field.field.choices %}
+        <div class="radio{% if field.field.required %} form-required{% endif %}{% if field.errors %} has-error{% endif %}">
+            {% block form_field_label %}
+                <label for="{{ field.auto_id }}_{{ choice.0 }}">
+                    {% block form_field %}
+                        <input type="radio" name="{{ field.name }}" id="{{ field.auto_id }}_{{ choice.0 }}" value="{{ field.value }}" {% if field.value == choice.0 %}checked="checked"{% endif %}>
+                        {% comment %}
```
Comment (FinalAngel) [inline aldryn_forms/boilerplates/bootstrap3/templates/aldryn_forms/fields/radioselectfield.html]: isn't that fixed using ``{% if field.value == choice.0 %}`` ?

File: aldryn_forms/boilerplates/bootstrap3/templates/aldryn_forms/fields/textfield.html [hunk 10]
```diff
@@ -0,0 +1,23 @@
+{% load aldryn_forms_tags i18n %}
+
+{% if field %}
+    <div class="form-group{% if field.field.required %} form-required{% endif %}{% if field.errors %} has-error{% endif %}">
+        {% block form_field_label %}
+            <label for="{{ field.auto_id }}">{{ field.label }}
```
Comment (sergii-kulakovskyi) [inline aldryn_forms/boilerplates/bootstrap3/templates/aldryn_forms/fields/textfield.html]: Could you move `{{ field.label }}` to new line in all similar situations?

**Comment by sergii-kulakovskyi** [comment]
LGTM

**Comment by FinalAngel** [comment]
LGTM

**Comment by FinalAngel** [comment]
wohoooooo :+1:

**Pull request merged**